File: fathersDay/Dsa/Day 27 (1047. Remove All Adjacent Duplicates In String)/Solution.py
class Solution:
    def RemoveDuplicates(self, s: str) -> str:
        stack = []
        for char in s:
            if stack and stack[-1] == char:
                stack.pop()
            else:
                stack.append(char)
        return ''.join(stack)
solution = Solution()
s = "abbaca"
result = solution.RemoveDuplicates(s)
print(result)  # Output: "ca", as adjacent duplicates are removed

File: fathersDay/Dsa/Day 27 (1047. Remove All Adjacent Duplicates In String)/Solution.java
import java.util.Stack;

class Solution {
    public String RemoveDuplicates(String s) {
        Stack<Character> stack = new Stack<>();
        for (char c : s.toCharArray()) {
            if (!stack.isEmpty() && stack.peek() == c) {
                stack.pop();
            } else {
                stack.push(c);
            }
        }
        StringBuilder sb = new StringBuilder();
        for (char c : stack) {
            sb.append(c);
        }
        return sb.toString();
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        String s = "abbaca";
        String result = solution.RemoveDuplicates(s);
        System.out.println('"' + result + '"'); // Output: "ca"
    }
}
